simulations/NavierStokes_2D/scripts/preprocess_posteriori.jl

```diff
@@ -30,4 +31,8 @@ io_post[ig].t[2, :]
 # * Create dataloader containing trajectories with the specified nunroll
 using CoupledNODE.NavierStokes: create_dataloader_posteriori
 nunroll = 5
-dataloader_posteriori = create_dataloader_posteriori(io_post[ig]; nunroll = nunroll, rng)
+dataloader_posteriori = create_dataloader_posteriori(io_post; nunroll = nunroll, rng)
+
+# Load the test data
+test_data = load("simulations/NavierStokes_2D/data/data_test.jld2", "data_test")
+test_io_post = create_io_arrays_posteriori(test_data, setups)
```

Comment:
This leads to a problem. Indeed `create_io_arrays_posteriori ` considers an array with multiple `NamedTuple` in it, each one representing a sample. 
For the *train* dataset, we have multiple samples, but for the *test* dataset we only have one sample and so `test_data` is directly a `NamedTuple` and not an `Array`.

I think we should not try to have special if statements to check the nature of the data here, but in `generate_data.jl` make sure that the format is `Array` even when there is a single sample. What do you think?

Reply:
Outdated comment. In `generate_data.jl` the *test* data is also an `Array`.